### PR TITLE
Highlight config changes

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #2296. Fixes: #1495.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- add a charm to the canvas
- open the inspector config
- make a change to the config
- the save/cancel buttons should appear
- the field should get a blue border
- the select box, toggle etc. should all do the same

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7267/

File: jujugui/static/gui/src/app/components/inspector/config/config.js [hunk 1]
```diff
@@ -144,22 +154,13 @@ YUI.add('inspector-config', function() {
...
-          configValues[activeRef.props.option.key] = activeRef.state.value;
-        }
-      });
       // The service name component is only shown if it's a ghost service.
       var serviceName = this.refs.ServiceName;
       var props = this.props;
       if (serviceName) {
         var service = props.service;
         var nameValue = serviceName.state.value;
+        console.log(nameValue);
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector/config/config.js]: left in log

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7284/

**Comment by huwshimi** [comment]
@hatched for "When I save a change then open the inspector again, it doesn't show as modified." I was thinking that the uncommitted changes (and the buttons appearing) are there to hint to the user that they need to save changes.

I guess there's the other part about showing everything in blue that has changed since the last deploy. In that case, is that even possible? Would we have to check the ECS for changes or something?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7285/

**Comment by jaycee** [comment]
QA notes:

Following with hatch's QA issues:

* "When I modify the name, the field changes as expected." -- Works, looks OK.
* "When I modify beats_port then undo the change, it still shows it as modified. This appears to only happen on "int" fields, so likely because its type is changing." -- Modifying an int field (udp_port) works as expected. Looks OK.
* "Whenever a field becomes 'modified' I get two scrollbars as it appears something re-renders and scrolls when the buttons become visible." -- I see the scrollbar move, but that's to be expected as the vertical space of the component changes when the buttons render. When I dismiss the buttons the scrollbar reverts as expeced. Looks OK.
* "When I save a change then open the inspector again, it doesn't show as modified."  -- When I modify a value and then save and close the inspector, the value is preserved when I next open it.

I'ms seeing correct highlights &c as described in the initial directions.

However, as I modify values in the inspector I occasionally get a bunch of garbage in it:
2017-05-05-115117_389x570_scrot

The more I type/modify in any field, the more this is generated.

I'll note however, this seems to exist in develop. I'll file a bug shortly, but it's not a QA blocker here.

QA OK.

**Comment by huwshimi** [comment]
Thanks @jaycee, I've fixed that escaping bug in: #2863.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7323/

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge/2344

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**